providers/aws-ec2/src/main/java/org/jclouds/aws/ec2/features/VPCApi.java

```diff
@@ -0,0 +1,114 @@
+import org.jclouds.aws.ec2.options.CreateVpcOptions;
+import org.jclouds.aws.ec2.xml.DescribeVPCsResponseHandler;
+import org.jclouds.aws.ec2.xml.ReturnValueHandler;
+import org.jclouds.aws.ec2.xml.VPCHandler;
+import org.jclouds.aws.filters.FormSigner;
+import org.jclouds.aws.ec2.binders.BindVpcIdsToIndexedFormParams;
+import org.jclouds.javax.annotation.Nullable;
+import org.jclouds.location.functions.RegionToEndpointOrProviderIfNull;
+import org.jclouds.rest.annotations.BinderParam;
+import org.jclouds.rest.annotations.EndpointParam;
+import org.jclouds.rest.annotations.Fallback;
+import org.jclouds.rest.annotations.FormParams;
+import org.jclouds.rest.annotations.RequestFilters;
+import org.jclouds.rest.annotations.VirtualHost;
+import org.jclouds.rest.annotations.XMLResponseParser;
+
+import com.google.common.collect.FluentIterable;
+
+/**
+ * Provides access to VPC Services.
+ * <p/>
+ */
+@RequestFilters(FormSigner.class)
+@VirtualHost
+public interface VPCApi {
+
+   /**
+    * Describes all of your VPCs
+    *
+    * @return VPCs or empty if there are none
+    * @see <a href=
+    *      "http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_DescribeVpcs.html"
+    *      >docs</href>
+    */
+   @Named("DescribeVpcs")
+   @POST
+   @Path("/")
+   @FormParams(keys = ACTION, values = "DescribeVpcs")
+   @XMLResponseParser(DescribeVPCsResponseHandler.class)
+   @Fallback(Fallbacks.EmptyFluentIterableOnNotFoundOr404.class)
+   FluentIterable<VPC> describeVpcsInRegion(
+           @EndpointParam(parser = RegionToEndpointOrProviderIfNull.class) @Nullable String region,
+           @BinderParam(BindVpcIdsToIndexedFormParams.class) String... vpcIds);
+
+   /**
+    * Creates a VPC with the specified CIDR block.
+    *
+    * @param region
+    *           VPCs are tied to the Region.
+    *           
+    * @param cidrBlock
+    *           The network range for the VPC, in CIDR notation. For example, 10.0.0.0/16.
+    * @return vpc
+    *
+    * @see <a href="http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpc.html"
+    *      />
+    * @see CreateVpcOptions
+    * @see <a href=
+    *      "http://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreateVpc.html"
+    *      />
+    */
+   @Named("CreateVpc")
+   @POST
+   @Path("/")
+   @FormParams(keys = ACTION, values = "CreateVpc")
+   @XMLResponseParser(VPCHandler.class)
+   VPC createVpc(
+           @EndpointParam(parser = RegionToEndpointOrProviderIfNull.class) @Nullable String region,
+           @FormParam("CidrBlock") String cidrBlock, CreateVpcOptions... options);
+   /**
+    * Deletes {@code VPC}.
+    *
+    * @param region
+    *           VPCs are tied to the Region where its files are located within Amazon S3.
+    * @param vpcId
+    *           The VPC ID.
+    *
+    */
+   @Named("DeleteVpc")
+   @POST
+   @Path("/")
+   @FormParams(keys = ACTION, values = "DeleteVpc")
+   @XMLResponseParser(ReturnValueHandler.class)
```

Comment:
Add a fallback to return false if the vpc does not exist? It is a common pattern in jclouds.